security/security.c

```diff
@@ -476,10 +519,28 @@ void __init security_add_hooks(struct security_hook_list *hooks, int count,
 				char *lsm)
 {
 	int i;
+	struct static_slot *slot;
 
 	for (i = 0; i < count; i++) {
 		hooks[i].lsm = lsm;
 		hlist_add_tail_rcu(&hooks[i].list, hooks[i].head);
+
+		if (!hooks[i].slots || !hooks[i].hook.generic_func)
+			continue;
+		for (slot = hooks[i].slots;
+		     slot < hooks[i].slots + SLOT_COUNT; slot++) {
+			if (!static_key_enabled(&slot->key->key)) {
+				__static_call_update(
+					slot->call_key, slot->call_tramp,
+					hooks[i].hook.generic_func);
+				static_branch_enable(slot->key);
+				break;
+			}
+		}
+		if (slot == hooks[i].slots + SLOT_COUNT) {
+			printk(KERN_ERR 
+			       "No slot remaining to add LSM hook %s\n", lsm);
```

Comment:
Already discussed but just to note down: we need to make this impossible.

If we keep the loop above (as opposed to setting up fixed LSM->slot index mapping), we'll want to keep this case in the code but, I think, use the `BUG()` macro to panic if we hit it.